src/lib/pubkey/dilithium/dilithium_common/dilithium_symmetric_primitives.h

```diff
@@ -61,8 +62,10 @@ class Dilithium_Symmetric_Primitives {
       enum class XofType { k128, k256 };
 
    protected:
-      Dilithium_Symmetric_Primitives(size_t commitment_hash_length_bytes) :
-            m_commitment_hash_length_bytes(commitment_hash_length_bytes) {}
+      Dilithium_Symmetric_Primitives(const DilithiumConstants& mode) :
+            m_commitment_hash_length_bytes(mode.commitment_hash_full_bytes()),
+            m_public_key_hash_bytes(mode.public_key_hash_bytes()),
+            m_mode(mode.mode()) {}
```

Comment:
Do we actually need to copy those constants? Couldn't we just ask `m_mode` when they are needed?

Reply:
Note that m_mode is not a `DilithiumConstants` object but simply the (much smaller) `DilithiumMode` object. We would need to store the DilithiumConstants object, which is significantly bigger (but not huge).